share/man/man1/brew.1

```diff
@@ -554,10 +554,6 @@ HOMEBREW_BROWSER
 If set, uses this setting as the browser when opening project homepages, instead of the OS default browser\.
 .
 .TP
-HOMEBREW_BUILD_FROM_SOURCE
```

Comment:
> I'd rather we didn't encourage it.

Personally, I think we should still document. The only thing should be undocumented should be either private or deprecated. Which neither of them apply to this feature. 

>  I don't think the use-case is valid 

I disagree. I think self brewing is one of base features homebrew should offer. Because there are some good reasons to do this, e.g. people who think highly of security than average users would like to build everything from source. Or users who have bad network connection to binary CDN would also like such feature. 

> we've already had issues from people using it.

If users have issues on simply building from source. Then I think this is a bug to solve rather to hide.